### DISCUSSION TITLE
Build system questions

### DISCUSSION THREAD

**Discussion by MatanZ** [transmission/transmission#2857] [Q&A] [answered]
Is it possible to build only the daemon, or only the utils, without editing CMakeLists.txt?

Is it possible to avoid clang-tidy during build without editing CMakeLists.txt? Building with clang-tidy takes literally hours on my system, and less than a minute without it.

The problem with modifying CMakeLists.txt is that it keeps resetting when I checkout branches, so I find myself spending a lot of time messing with this file.

**Comment by mikedld** [comment, marked as answer]
CMake has a notion of "options" (and "cache variables") which you can set to customize the generated build system. The options currently exposed by Transmission could be found in the top-level CMakeLists.txt: https://github.com/transmission/transmission/blob/1f15933fda9894d5ec5b86089fe65bc73c9d4029/CMakeLists.txt#L23-L50

In your case, you'd need to pass several of them while configuring, e.g. for daemon only:
```sh
cmake /path/to/source/dir \
    -DENABLE_GTK=OFF \
    -DENABLE_QT=OFF \
    -DENABLE_MAC=OFF \
    -DENABLE_UTILS=OFF \
    -DRUN_CLANG_TIDY=OFF
```

**Reply by MatanZ** [reply]
Thanks. Being more familiar with make, I was looking for a solution there, so I did not check the possibility of cmake command line.
